packages/Polyfill/src/FeatureSupport/FunctionSupportResolver.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Polyfill\FeatureSupport;
+
+use Rector\Php\PhpVersionProvider;
+
+final class FunctionSupportResolver
+{
+    /**
+     * @var string[][]
+     */
+    private const FUNCTIONS_BY_VERSION = [
+        '5.6' => ['session_abort'],
+    ];
+
+    /**
+     * @var PhpVersionProvider
+     */
+    private $phpVersionProvider;
+
+    public function __construct(PhpVersionProvider $phpVersionProvider)
+    {
+        $this->phpVersionProvider = $phpVersionProvider;
+    }
+
+    public function isFunctionSupported(string $desiredFunction): bool
+    {
+        foreach (self::FUNCTIONS_BY_VERSION as $version => $functions) {
+            foreach ($functions as $function) {
+                if ($desiredFunction !== $function) {
+                    continue;
+                }
+
+                if (! $this->phpVersionProvider->isAtLeast($version)) {
```

Comment:
I guess it would also be helpfull to remove code when explicit php version checks are used (might be a separTe rector then?)

In codes like https://github.com/redaxo/redaxo/blob/591146a1dc60e8aacefd58dc9b7e9c307c0983b9/redaxo/src/addons/tests/vendor/phpspec/prophecy/src/Prophecy/Doubler/Generator/ClassCodeGenerator.php#L85

Reply:
Sure! I though of that too.

Could you help by making a new issue with minimal code (3-lines) in diff a format?

Reply:
See https://github.com/rectorphp/rector/issues/2573 https://github.com/rectorphp/rector/issues/2574